cocotb/decorators.py

```diff
@@ -456,12 +457,18 @@ def my_test(dut):
         stage (int, optional)
             Order tests logically into stages, where multiple tests can share a stage.
     """
+    _id_count = 0
+
     def __init__(self, f, timeout_time=None, timeout_unit=None,
                  expect_fail=False, expect_error=False,
                  skip=False, stage=None):
 
+        self._id = self._id_count
+        self._id_count += 1
```

Comment:
I think this does the wrong thing, and you want
```suggestion
        type(self)._id_count += 1
```